skopt/plots.py

```diff
@@ -410,5 +484,227 @@ def plot_evaluations(result, bins=20, dimensions=None):
+    """Create and return a Matplotlib figure with a landscape
+    contour-plot of the last fitted model of the search-space,
+    overlaid with all the samples from the optimization results,
+    for the two given dimensions of the search-space.
+
+    Parameters
+    ----------
+    * `result` [`OptimizeResult`]
+        The optimization results e.g. from calling `gp_minimize()`.
+
+    * `dimension_id1` [int or str]:
+        Either an index or name for a dimension in the search-space.
+
+    * `dimension_id2` [int or str]:
+        Either an index or name for a dimension in the search-space.
+
+    * `n_samples` [int, default=250]
+        Number of random samples used for estimating the contour-plot
+        of the objective function.
+
+    * `n_points` [int, default=40]
+        Number of points along each dimension where the partial dependence
+        is evaluated.
+
+    * `levels` [int, default=10]
+        Number of levels to draw on the contour plot.
+
+    * `zscale` [str, default='linear']
+        Scale to use for the z axis of the contour plots. Either 'log'
+        or linear for all other choices.
+
+    Returns
+    -------
+    * `fig`: [`matplotlib.figure.Figure`]:
+        The Matplotlib Figure-object.
+        For example, you can save the plot by calling `fig.savefig('file.png')` 
+    """
+
+    # Get the search-space instance from the optimization results.
+    space = result.space
+
+    # Get the dimension-object, its index in the search-space, and its name.
+    dimension1, index1, dimension_name1 = space.get_dimension(id=dimension_id1)
+    dimension2, index2, dimension_name2 = space.get_dimension(id=dimension_id2)
+
+    # Ensure dimensions are not Categorical.
+    if type(dimension1) == Categorical or type(dimension2) == Categorical:
+        raise ValueError("Categorical dimension is not supported.")
+
+    # Get the samples from the optimization-log for the relevant dimensions.
+    samples1 = _get_samples_dimension(result=result, index=index1)
+    samples2 = _get_samples_dimension(result=result, index=index2)
+
+    # Get the best-found samples for the relevant dimensions.
+    best_sample1 = result.x[index1]
+    best_sample2 = result.x[index2]
+
+    # Get the last fitted model for the search-space.
+    last_model = result.models[-1]
+
+    # Get new random samples from the search-space and transform if necessary.
+    sample_points = space.rvs(n_samples=n_samples)
+    sample_points = space.transform(sample_points)
+
+    # Estimate the objective function for these sampled points
+    # using the last fitted model for the search-space.
+    xi, yi, zi = partial_dependence(space=space,
+                                    model=last_model,
+                                    i=index2, j=index1,
+                                    sample_points=sample_points,
+                                    n_points=n_points)
+
+    # Start a new plot.
+    fig = plt.figure()
```

Comment:
we should create a figure and axis. Then later we can do `ax.plot(...)` and `ax.scatter()` instead of using `plt.scatter`.

Reply:
I never really understood the semantics of Matplotlib. What is the reason for this?

Reply:
The `plt.plot` and co interface uses global state and is meant to make interactive plotting easy. The object orientated interface has less potential for weird interactions between plots because it does not have global state. The best way to learn is probably following some matplotlib tutorials/videos and looking at how other libraries that do plotting handle things.

Reply:
OK, I see. I'll take a look if I can fix this. I've used Matplotlib many times for many projects, but never liked its API.